src/controllers/restful/PaymentController.ts

```diff
@@ -5,221 +5,318 @@
 
-    // Check PagoPa response content.
+    // Check PagoPA response content.
     // If it contains an error, an HTTP error will be provided to BackendApp
     if (
-      paymentActivationPagoPaResponse.nodoAttivaRPTRisposta.esito ===
-      PPTPortTypes.Esito.KO
+      iNodoAttivaRPTOutput.nodoAttivaRPTRisposta.esito === PPTPortTypes.Esito.KO
     ) {
       return ResponseErrorInternal("Error during payment check: esito === KO");
     }
 
-    // Convert the output provided by PagoPa (SOAP response)
+    // Convert the output provided by PagoPA (SOAP response)
     // to a BackendApp response (RESTful response), mapping the result information.
     // Send a response to BackendApp
-    return PaymentsConverter.getPaymentsActivationResponse(
-      paymentActivationPagoPaResponse
+    return PaymentsConverter.getPaymentActivationsPostResponse(
+      iNodoAttivaRPTOutput
     ).fold<
       | IResponseErrorValidation
-      | IResponseSuccessJson<PaymentsActivationResponse>
+      | IResponseSuccessJson<PaymentActivationsPostResponse>
     >(
-      ResponseErrorFromValidationErrors(PaymentsActivationResponse),
+      ResponseErrorFromValidationErrors(PaymentActivationsPostResponse),
       ResponseSuccessJson
     );
   };
 }
 
-/** Receive an async activation result frop PagoPA
- * TODO: [#157910857] Creazione dei controller SOAP per l'esposizione dei servizi verso PagoPa
- * TODO: [#158176380] Gestione della conferma di attivazione di un pagamento
+/**
+ * This controller is invoked by PagoPA that provides a paymentId
+ * related to a previous async request (attivaRPT)
+ * It just store this information into redis db. This information will be retrieved by App using polling
+ * @param {IcdInfoPagamentoInput} cdInfoPagamentoInput - The request from PagoPA
+ * @param {RedisTimeout} redisTimeout - The expiration timeout for the information to store
+ * @param {RedisClient} redisClient - The redis client used to store the paymentId
+ * @param {(cdInfoPagamentoOutput: IcdInfoPagamentoOutput) => void} callback - Callback function to send a feedback to PagoPA
+ * @return {Promise<IResponse*>} The response content to send to applicant
  */
-// tslint:disable-next-line:no-empty
-export async function notifyPaymentStatusToAPINotifica(): Promise<void> {}
+export function updatePaymentActivationStatusIntoDB(
+  cdInfoPagamentoInput: IcdInfoPagamentoInput,
+  redisTimeout: number,
+  redisClient: redis.RedisClient,
+  callback: (cdInfoPagamentoOutput: IcdInfoPagamentoOutput) => void
+): void {
+  // Check DB connection status
+  if (redisClient.connected !== true) {
+    callback({
+      esito: Esito.KO
+    });
+  }
+  const setAsyncRedis = promisify(redisClient.set).bind(redisClient);
+  setAsyncRedis(
+    cdInfoPagamentoInput.codiceContestoPagamento,
+    cdInfoPagamentoInput.idPagamento,
+    "EX", // Set the specified expire time, in seconds.
+    redisTimeout
+  ).catch((error: Error) => {
+    return error;
```

Comment:
what's the purpose of this return here ? it looks like it's unused. return a Promise<Either<Error, ResultType> 
from this method (updatePaymentActivationStatusIntoDB). Do not take the callback as input, let the caller specify what's next calling .then()